### PR TITLE
Realize inputs to DynamicScalar before unwrapping storage

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118411
* __->__ #118125
* #117862

Fixes https://github.com/pytorch/pytorch/issues/118102

Unfortunately, the test still fails due to an unrelated problem https://github.com/pytorch/pytorch/issues/117665

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -2611,6 +2611,8 @@ def inner(
         dtype = dtype or torch.get_default_dtype()
         if len(size) == 1 and isinstance(size[0], (list, tuple, torch.Size)):
             size = tuple(size[0])
+        for s in size:
+            assert not isinstance(s, torch.SymInt)
```
Comment (ezyang) [inline torch/_inductor/lowering.py:2615]: This makes the later problem trigger more clearly @eellison

Comment (eellison) [reply torch/_inductor/lowering.py:2615]: Maybe add a comment or update assertion along the lines of `should be Symbol not SymInt`

**Review by eellison — APPROVED** [review]

**Review by jansel — DISMISSED** [review]
Test failures?

**Comment by ezyang** [comment]
@jansel just a dumb typo for the expectedFailure lol

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**